src/quo/context.cljs

```diff
@@ -0,0 +1,22 @@
+(ns quo.context
+  (:require
+    ["react" :as react]
+    [oops.core :as oops]
+    [react-native.core :as rn]))
+
+(defonce ^:private context (react/createContext {:theme :light :screen-id nil}))
+
+(defn provider
+  [data & children]
+  (into [:> (.-Provider context) {:value data}]
+        children))
+
+(defn use-theme
+  "A hook that returns the current theme keyword."
+  []
+  (keyword (oops/oget (rn/use-context context) :theme)))
+
+(defn use-screen-id
+  "A hook that returns the current screen id."
+  []
+  (keyword (oops/oget (rn/use-context context) :screenId)))
```

Comment:
Hi @ilmotta @flexsurfer, PR is still in WIP. Pinging you guys to notify the solution I have opted for.